### PR TITLE
Factor out common cmake code to moveit_package macro

### PR DESCRIPTION
**Pull request by lilustga**
### Description

Created the moveit_common package containing a moveit_package() macro. This is used to apply basic cmake code to all moveit packages.

This macro allows CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS to be set for all moveit packages and allows for any new cmake code to be applied to all packages easily in the future.

Similar idea to this commit in nav2:
ms-iot/navigation2@3f65576

### Checklist
- [ ] **Required by CI**: Code is auto formatted using [clang-format](http://moveit.ros.org/documentation/contributing/code)
- [ ] Extend the tutorials / documentation [reference](http://moveit.ros.org/documentation/contributing/)
- [ ] Document API changes relevant to the user in the [MIGRATION.md](https://github.com/ros-planning/moveit/blob/master/MIGRATION.md) notes
- [ ] Create tests, which fail without this PR [reference](https://ros-planning.github.io/moveit_tutorials/doc/tests/tests_tutorial.html)
- [ ] Include a screenshot if changing a GUI
- [ ] While waiting for someone to review your request, please help review [another open pull request](https://github.com/ros-planning/moveit/pulls) to support the maintainers

[//]: # "You can expect a response from a maintainer within 7 days. If you haven't heard anything by then, feel free to ping the thread. Thank you!"

### CODE REVIEW

**Comment by lilustga** [comment]
Looks like clang format and gcc build have passed however there's a warning from moveit_planners_ompl causing clang build to fail.

The warning is 'ompl_interface::StateValidityChecker::isValid' hides overloaded virtual function [-Woverloaded-virtual]

It could be suppressed from the moveit_planners_ompl CMakeLists.txt, or should this be fixed?

**Comment by henningkayser** [comment]
> Looks like clang format and gcc build have passed however there's a warning from moveit_planners_ompl causing clang build to fail.
> 
> The warning is 'ompl_interface::StateValidityChecker::isValid' hides overloaded virtual function [-Woverloaded-virtual]
> 
> It could be suppressed from the moveit_planners_ompl CMakeLists.txt, or should this be fixed?

@lilustga Well, this is strange. I don't know how this issue could have been introduced here, it should be checking for the same warnings as before. I think it should be fixed if possible.

File: moveit_core/CMakeLists.txt
Comment (henningkayser) [inline moveit_core/CMakeLists.txt:20]: Please remove

File: moveit_demo_nodes/run_moveit_cpp/CMakeLists.txt
Comment (henningkayser) [inline moveit_demo_nodes/run_moveit_cpp/CMakeLists.txt:10]: Shouldn't we just add this and the `moveit_package()` macro to the top in all CMakeLists files?

**Review by henningkayser — APPROVED** [review]
Looks good to me. Please consider moving the moveit_common macros to the top with a short comment, I think it would be cleaner and more consistent.

**Review by tylerjw — APPROVED** [review]
I approve this approach.  It is cleaner and a nice way to consolidate common cmake settings for the whole repo.  Please remove the commented out code.

**Comment by lilustga** [comment]
Travis clang build and code coverage look like they failed due to a transient issue. @henningkayser are you able to restart the build checks (I think it requires write access)?

**Comment by henningkayser** [comment]
> Travis clang build and code coverage look like they failed due to a transient issue. @henningkayser are you able to restart the build checks (I think it requires write access)?

It seems like the checks work now, but there are some unrelated CI fixes I'm currently working on

**Comment by henningkayser** [comment]
CI issues are fixed with #294. After that, I would squash-merge this PR

**Pull request merged**
